### PR TITLE
gadget: multi line support in gadget's cmdline file

### PR DESCRIPTION
**Pull request by bboozzoo**
Support a multiline format of the cmdline file shipped by the gadget. Now it is valid to have a content like this:
```
# comment
paramter parameter
# more
# comments
panic=-1
```

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -176,7 +176,7 @@ backends:
             - ubuntu-20.04-64:
                   image: ubuntu-2004-64-virt-enabled
                   storage: 20G
-                  workers: 6
+                  workers: 7
```
Comment (mvo5) [inline spread.yaml:179]: While I agree with the change here it looks a bit unrelated. It's a separate commit so probably ok

**Review by mvo5 — APPROVED** [review]
Thank you

File: gadget/gadget.go [hunk 2]
```diff
@@ -1048,16 +1050,53 @@ func KernelCommandLineFromGadget(gadgetDirOrSnapPath string) (cmdline string, fu
...
+	if err := s.Err(); err != nil {
+		return "", err
+	}
+	kargs, err := osutil.KernelCommandLineSplit(filtered.String())
+	if err != nil {
+		return "", err
 	}
 	for _, argValue := range kargs {
+		if strings.HasPrefix(argValue, "#") {
+			return "", fmt.Errorf("incorrect use of # in argument %q", argValue)
```
Comment (pedronis) [inline gadget/gadget.go:1094]: s/incorrect use /unexpected or invalid use/

**Review by pedronis — APPROVED** [review]
thank you, small comment about the error

**Comment by bboozzoo** [comment]
The spread failures, specifically those in 20.04 nested test are unrelated to the PR. @mvo5 can you land it?

**Review by anonymouse64 — COMMENTED** [review]
post merge lgtm from me thanks for this
